lib/Qpsmtpd/SMTP.pm

```diff
@@ -681,21 +681,21 @@ sub data_respond {
 
     my $timeout = $self->config('timeout');
     while (defined($_ = $self->getline($timeout))) {
-        if ($_ eq ".\r\n") {
-            $complete++;
-            $_ = '';
-        }
-        $i++;
-
         # Reject messages that have either bare LF or CR. rjkaes noticed a
         # lot of spam that is malformed in the header.
 
-        if ($_ eq ".\n" || $_ eq ".\r") {
+        if ($_ !~ /\r\n$/) {
```

Comment:
Actually, why isn't \n$ allowed inside the message body? It wouldn't be a new line, technically, just a \n inside a message.

I thought it should only be disallowed as a "end of message" marker.

Reply:
RFC 5322 (2008), Section 4: "Finally, certain characters that were formerly allowed in messages appear in this section.... CR and LF were allowed to appear in messages other than as CRLF;"

Appendix B.27: "Free CR and LF not allowed."